pkg/auth/api/user/user_actions.go

```diff
@@ -170,3 +188,22 @@ func (h *Handler) refreshAttributes(actionName string, action *types.Action, req
 func (h *Handler) userCanRefresh(request *types.APIContext) bool {
 	return request.AccessControl.CanDo(v3.UserGroupVersionKind.Group, v3.UserResource.Name, "create", request, nil, request.Schema) == nil
 }
+
+func getMinimumPasswordLength() int {
+	return settings.PasswordMinLength.GetInt()
+}
+
+// validatePassword will ensure a password is the minimum required length, and that the password does not contain the username.
+func validatePassword(user string, pass string, minPassLen int) error {
+	passLen := len(pass)
+
+	if passLen < minPassLen {
+		return errors.Errorf("password must be at least %v characters", minPassLen)
```

Comment:
Does this get properly displayed to user ? Is there any UI work that would need to get done ?

Reply:
The error is passed back in a response to the UI. It may make sense to capitalize the error, although I'm not sure if we should format from backend, frontend, etc., may need UI opinion.

Reply:
Follow common convention in backend code is fine, I was primarily concerned that the error already displays correctly around the login box.

Reply:
![error-format](https://user-images.githubusercontent.com/33796120/135898893-c1e34153-d7c7-4e33-859a-a961d9b52493.png)
The error displays below the continue button (and appears to resize correctly on chrome at least).

Reply:
Lets capitalize password, then looks good

Reply:
Ok, I'll fix capitlization and I'm adding here so that the behavior is the same when creating users in "Users and Authentication".

https://github.com/rancher/rancher/blob/1874d541a26721d3ee9d657bf2c93092b1fa54ce/pkg/auth/api/user/user_store.go#L109-L117

Reply:
ahh good catch, you should double check if anything else sets it, should be able to track back via code.